src/ansys/fluent/core/utils/file_transfer_service.py

```diff
@@ -139,26 +141,46 @@ class RemoteFileTransferStrategy(FileTransferStrategy):
 
     def __init__(
         self,
+        image: Optional[str] = None,
+        ports: Optional[dict] = None,
         container_mount_path: Optional[str] = None,
         host_mount_path: Optional[str] = None,
     ):
-        self.host_port = random.randint(5000, 6000)
+        self.docker_client = docker.from_env()
+        self.image = image if image else "ghcr.io/ansys/tools-filetransfer:latest"
         self.container_mount_path = (
             container_mount_path if container_mount_path else "/home/container/workdir/"
         )
         self.host_mount_path = (
             host_mount_path if host_mount_path else pyfluent.EXAMPLES_PATH
         )
         try:
-            self.server = subprocess.Popen(
-                f"docker run -p {self.host_port}:50000 -v {self.host_mount_path}:{self.container_mount_path} ghcr.io/ansys/tools-filetransfer:latest",
-                shell=True,
+            self.host_port = random.randint(5000, 6000)
+            self.ports = ports if ports else {"50000/tcp": self.host_port}
+            self.server = self.docker_client.containers.run(
+                image=self.image,
+                ports=self.ports,
+                detach=True,
+                volumes={
+                    self.host_mount_path: {
+                        "bind": self.container_mount_path,
+                        "mode": "rw",
+                    }
+                },
             )
         except Exception:
             self.host_port = random.randint(6000, 7000)
-            self.server = subprocess.Popen(
-                f"docker run -p {self.host_port}:50000 -v {self.host_mount_path}:{self.container_mount_path} ghcr.io/ansys/tools-filetransfer:latest",
-                shell=True,
+            self.ports = ports if ports else {"50000/tcp": self.host_port}
+            self.server = self.docker_client.containers.run(
```

Comment:
Minor: I think `container` would be a better variable name
```suggestion
            self.container = self.docker_client.containers.run(
```

Reply:
Done. Thank you.